src/components/chip.tsx

```diff
@@ -65,23 +70,24 @@ export const Chip: FC<ChipProps & HTMLAttributes<HTMLButtonElement>> = ({
   );
 
   return (
-    <button
-      type="button"
+    <span
+      role="button"
       className={cn(
         'chip',
         { 'chip--disabled': disabled, 'chip--compact': compact },
         className
       )}
-      title={title}
+      onKeyPress={onKeyPress}
       onClick={onClick}
-      disabled={typeof disabled === 'boolean' ? disabled : !onClick}
+      title={title}
+      tabIndex={0}
```

Comment:
I would prefer if the role of button is only applied if onClick and/or onKeyPress is set. Same thing for tabIndex.
I might lean towards checking if onClick exists, and if so render a button like before, and if not render a plain span, this is in order to avoid having to use roles and tab indices.

Reply:
Can do - just wondering what is the drawback of using roles and tab indices?

Reply:
Not really a direct drawback, it's more to follow the first rule of ARIA which is ~don't talk about ARIA~ don't use ARIA if you can avoid it https://www.w3.org/TR/using-aria/#rule1

Reply:
Had a go at doing this. Let me know what you think and if OK I'll update the unit tests to get more coverage.